### PR TITLE
refactor(function analysis)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Refactoring of the function analysis

### Added

- Filter for the started_at attribute
- Test for filter mentioned above

### Changed

- Function name `analysis` to `get_analysis_by_case_entry_id_and_started_at

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by seallard — APPROVED** [review]
Looks good 👍

File: cg/store/api/find_business_data.py [hunk 1]
```diff
@@ -113,9 +113,21 @@ def latest_analyses(self) -> Query:
...
-        return self.Analysis.query.filter_by(family=family, started_at=started_at).first()
+        filter_functions = [
+            AnalysisFilter.FILTER_BY_CASE_ENTRY_ID,
+            AnalysisFilter.FILTER_BY_STARTED_AT,
+        ]
+
+        return apply_analysis_filter(
+            analyses=self._get_query(Analysis),
+            case_entry_id=case_entry_id,
+            date=started_at,
```
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply cg/store/api/find_business_data.py:128]: fixed

**Comment by ChrOertlin** [comment]
cg generate delivery-report -d --analysis-started-at '2023-03-26 17:45:29' clearsnail

Using analysis started at: 2023-03-26 17:45:29

**Comment by ChrOertlin** [comment]
Stage:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/housekeeper-bundles
done.
Log deploy... done.
cg, version 27.1.38

Production:
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/housekeeper-bundles
done.
Log deploy... done.
cg, version 27.1.38

**Pull request merged**